portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -1,21 +1,47 @@
 package com.google.sps.servlets;
 
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
 
 @WebServlet("/posts")
 public class FormHandlerServlet extends HttpServlet {
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the values entered in the form. Sanitize user input to remove HTML tags and JavaScript.
+    String senderName = Jsoup.clean(
+      request.getParameter("sender-name"), 
+      Whitelist.none()
+    );
+    String senderOrganization = Jsoup.clean(
+      request.getParameter("sender-organization"), 
+      Whitelist.none()
+    );
+    String senderMessage = Jsoup.clean(
+      request.getParameter("sender-message"), 
+      Whitelist.none()
+    );
 
-    // Get the values entered in the form.
-    String senderName = request.getParameter("sender-name");
-    String senderOrganization = request.getParameter("sender-organization");
-    String senderMessage = request.getParameter("sender-message");
+    // Set up datastore object.
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    KeyFactory keyFactory = datastore.newKeyFactory().setKind("Post");
+    FullEntity postEntity =
+        Entity.newBuilder(keyFactory.newKey())
+            .set("sender-name", senderName)
+            .set("sender-organization", senderOrganization)
+            .set("sender-message", senderMessage)
```

Comment:
nit: Something to consider when you have the same String literal repeated multiple times in different places is to make it a constant, which can make later refactoring easier and help catch bugs earlier. For example, a typo in the String literal will be hard to track down but a typo in the constant name will be a compiler error.

Reply:
Thanks for pointing this out. It's definitely something I overlooked. I will change the String literals to constants.